app/controllers/ems_infra_dashboard_controller.rb

```diff
@@ -0,0 +1,36 @@
+class EmsInfraDashboardController < ApplicationController
+  extend ActiveSupport::Concern
+
+  before_action :check_privileges
+  before_action :get_session_data
+  after_action :cleanup_action
+  after_action :set_session_data
+
+  def show
+    if params[:id].nil?
+      @breadcrumbs.clear
+    end
+  end
+
+  def index
+    redirect_to :action => 'show'
+  end
+
+  def data
+    render :json => {:data => collect_data(params[:id])}
+  end
+
+  private
+
+  def get_session_data
```

Comment:
same with `{get|set}_session_data` can we just avoid these methods in new controllers?